### PR TITLE
make schedule cache list fast

### PR DESCRIPTION
**Pull request by HardySimpson**
**Reduce memory allocation at runtime of scheduler cache**:

this PR help reduce memeory allocation with to much slice append

**Release note**:

```
none
```

### CODE REVIEW

**Comment by k82cn** [comment]
@HardySimpson , thanks very much for your PR :). Do you have any data that shown the improvement?

/cc @kubernetes/sig-scheduling-pr-reviews

**Comment by jayunit100** [comment]
Also check https://github.com/kubernetes/community/blob/master/contributors/devel/development.md#is-this-a-performance-improvement- 

for generic guidelines on performance improvement PRs

**Comment by timothysc** [comment]
Simple A-B results should suffice.

File: plugin/pkg/scheduler/schedulercache/cache.go [hunk 1]
```diff
@@ -95,7 +95,7 @@ func (cache *schedulerCache) UpdateNodeNameToInfoMap(nodeNameToInfo map[string]*
 func (cache *schedulerCache) List(selector labels.Selector) ([]*v1.Pod, error) {
 	cache.mu.Lock()
 	defer cache.mu.Unlock()
-	var pods []*v1.Pod
+	pods := make([]*v1.Pod, 0, len(cache.podStates))
```
Comment (bsalamat) [inline plugin/pkg/scheduler/schedulercache/cache.go]: This allocates memory enough to hold all the pods, but in practice only a fractions of pods are probably matched against the selector and added to the array. So, the new code probably wastes memory, which may or may not be justifiable given the performance improvement. That brings up back to the point that other reviewers mentioned as well. How much performance improvement do we get out of this change?

**Comment by jayunit100** [comment]
Echoing the other reviewers, copied over from the Performance PR guidelines above. 
```
    - A golang Benchmark test.
    - A visual depiction of reduced metric load on a cluster (measurable using metrics/ endpoints and grafana).
    - A hand-instrumented timing test (i.e. adding some logs into the controller manager).
```
One of those results should be submitted with this....  otherwise maybe close the PR (temporarily, until  the performance effect is measurable)

**Comment by davidopp** [comment]
+1 for demonstration of performance impact before we agree to take this.

**Pull request closed without merging**